pilot/pkg/security/authz/model/matcher/string.go

```diff
@@ -27,7 +27,16 @@ func StringMatcher(v string, treatWildcardAsRequired bool) *envoy_matcher.String
 
 // StringMatcherRegex creates a regex string matcher for regex.
 func StringMatcherRegex(regex string) *envoy_matcher.StringMatcher {
-	return &envoy_matcher.StringMatcher{MatchPattern: &envoy_matcher.StringMatcher_Regex{Regex: regex}}
+	return &envoy_matcher.StringMatcher{
+		MatchPattern: &envoy_matcher.StringMatcher_SafeRegex{
+			SafeRegex: &envoy_matcher.RegexMatcher{
+				EngineType: &envoy_matcher.RegexMatcher_GoogleRe2{
```

Comment:
should we use the engine defined here: https://github.com/istio/istio/blob/a4ce3ded34e35ec82ff564fe3bf2cf3d82dcaa20/pilot/pkg/networking/core/v1alpha3/route/route.go#L67? I think there were some concerns about the max program size being too small by default

Reply:
The authZ policy doesn't expose the regex in the API, we only use it internally to generate the `.+` expression for non-empty matching. So the default value (100) should be much enough for us.

Reply:
oh, cool.